main.cpp

```diff
@@ -19,13 +19,29 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
     void run(const MatchFinder::MatchResult &Result) override {

```

Comment:
- How to prevent parentheses from being added if they are already present in the (int)(f) code?
- How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?

Reply:
Использовать !isa
Напремер:
`if (!isa<ParenExpr>(exprecion->getSubExprAsWritten()->IgnoreImpCasts())) {
      new_pos.append("(");
}
auto new_expr = Lexer::getLocForEndOfToken(expr->getEndLoc(), 0, sourse, LangOptions());
    
if (!isa<ParenExpr>(exprecion->getSubExprAsWritten()->IgnoreImpCasts())) {
      _rewriter.InsertText(new_expr, ")");
}`

Reply:
2. if (castExprPtr->getCastKind() == CK_ToVoid)